### PR TITLE
Navigation

### PR DESCRIPTION
**Pull request by duy007**
- [X] Added navigation for home, search, add, and options pages. 
- [X] Added navigation for login and sign-up page
- [X] username and password stored in shared preferences after successful login
- [X] added a very scuff bottom bar (a drawback of not using fragments)
- [ ] turned off navigating to home page animation.

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -16,6 +16,9 @@ android {
 
         testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
     }
+    buildFeatures {
+        viewBinding = true
```
Comment (freymend) [inline app/build.gradle.kts:20]:
```suggestion
        viewBinding = true
        dataBinding = true
```

Comment (freymend) [reply app/build.gradle.kts:20]: Also add `id("org.jetbrains.kotlin.kapt")` to `plugins { }`

File: app/src/main/java/edu/uw/ischool/opensecrets/auth/SignUpActivity.kt [hunk 2]
```diff
@@ -74,7 +74,7 @@ class SignUpActivity : AppCompatActivity() {
                 runOnUiThread {
```
Comment (freymend) [inline app/src/main/java/edu/uw/ischool/opensecrets/auth/SignUpActivity.kt:74]: Run the intellij format code command

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -16,6 +16,10 @@
         android:supportsRtl="true"
         android:theme="@style/Theme.OpenSecrets"
         tools:targetApi="31">
+        <activity android:name=".OptionActivity"></activity>
+        <activity android:name=".EntryOverviewEditActivity"></activity>
+        <activity android:name=".EntryTextActivity"></activity>
+        <activity android:name=".SearchActivity"></activity>
```
Comment (freymend) [inline app/src/main/AndroidManifest.xml:22]:
```suggestion
        <activity android:name=".SearchActivity" />
```

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -16,6 +16,10 @@
         android:supportsRtl="true"
         android:theme="@style/Theme.OpenSecrets"
         tools:targetApi="31">
+        <activity android:name=".OptionActivity"></activity>
+        <activity android:name=".EntryOverviewEditActivity"></activity>
+        <activity android:name=".EntryTextActivity"></activity>
```
Comment (freymend) [inline app/src/main/AndroidManifest.xml:21]:
```suggestion
        <activity android:name=".EntryTextActivity" />
```

File: app/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -16,6 +16,10 @@
         android:supportsRtl="true"
         android:theme="@style/Theme.OpenSecrets"
         tools:targetApi="31">
+        <activity android:name=".OptionActivity"></activity>
+        <activity android:name=".EntryOverviewEditActivity"></activity>
```
Comment (freymend) [inline app/src/main/AndroidManifest.xml:20]:
```suggestion
        <activity android:name=".EntryOverviewEditActivity" />
```

File: app/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -16,6 +16,10 @@
         android:supportsRtl="true"
         android:theme="@style/Theme.OpenSecrets"
         tools:targetApi="31">
+        <activity android:name=".OptionActivity"></activity>
```
Comment (freymend) [inline app/src/main/AndroidManifest.xml:19]:
```suggestion
        <activity android:name=".OptionActivity" />
```

File: app/src/main/res/values/themes.xml
Comment (freymend) [inline app/src/main/res/values/themes.xml:1]: Revert this change

File: app/src/main/java/edu/uw/ischool/opensecrets/model/Entry.kt
Comment (freymend) [inline app/src/main/java/edu/uw/ischool/opensecrets/model/Entry.kt:1]: Run intellij format code command

File: app/src/main/java/edu/uw/ischool/opensecrets/auth/LoginActivity.kt
Comment (freymend) [inline app/src/main/java/edu/uw/ischool/opensecrets/auth/LoginActivity.kt:1]: Run intellij format code command

File: app/src/main/java/edu/uw/ischool/opensecrets/SearchActivity.kt
Comment (freymend) [inline app/src/main/java/edu/uw/ischool/opensecrets/SearchActivity.kt:1]: Run intellij format code command

File: app/src/main/java/edu/uw/ischool/opensecrets/OptionActivity.kt
Comment (freymend) [inline app/src/main/java/edu/uw/ischool/opensecrets/OptionActivity.kt:1]: Run intellij format code command

File: app/src/main/java/edu/uw/ischool/opensecrets/MainActivity.kt
Comment (freymend) [inline app/src/main/java/edu/uw/ischool/opensecrets/MainActivity.kt:1]: Run intellij format code command

File: app/src/main/java/edu/uw/ischool/opensecrets/EntryTextActivity.kt
Comment (freymend) [inline app/src/main/java/edu/uw/ischool/opensecrets/EntryTextActivity.kt:1]: Run intellij format code command

File: app/src/main/java/edu/uw/ischool/opensecrets/EntryOverviewEditActivity.kt
Comment (freymend) [inline app/src/main/java/edu/uw/ischool/opensecrets/EntryOverviewEditActivity.kt:1]: Run intellij format code command

File: app/src/main/java/edu/uw/ischool/opensecrets/MainActivity.kt [hunk 7]
```diff
@@ -1,22 +1,51 @@
...
 import android.content.Intent
-import android.os.Bundle
-import android.util.Log
+
 import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
 import edu.uw.ischool.opensecrets.auth.LoginActivity
 import java.io.File
+import android.util.Log
+import android.view.Menu
```
Comment (freymend) [inline app/src/main/java/edu/uw/ischool/opensecrets/MainActivity.kt:11]:
```suggestion
```
Unused import

File: app/src/main/java/edu/uw/ischool/opensecrets/MainActivity.kt [hunk 8]
```diff
@@ -1,22 +1,51 @@
...
+import android.view.MenuInflater
```
Comment (freymend) [inline app/src/main/java/edu/uw/ischool/opensecrets/MainActivity.kt:12]:
```suggestion
```

Unused import

File: app/src/main/java/edu/uw/ischool/opensecrets/model/Entry.kt [hunk 9]
```diff
@@ -0,0 +1,11 @@
+package edu.uw.ischool.opensecrets.model
+
+import kotlinx.coroutines.selects.SelectInstance
+import java.util.Date
+
+data class Entry(
+    val title: String,
+    val color: String,
+    val text: String,
+    val dataCreated: Date
```
Comment (freymend) [inline app/src/main/java/edu/uw/ischool/opensecrets/model/Entry.kt:10]:
```suggestion
    val dateCreated: Date
```

File: app/src/main/java/edu/uw/ischool/opensecrets/model/Entry.kt [hunk 10]
```diff
@@ -0,0 +1,11 @@
+package edu.uw.ischool.opensecrets.model
+
+import kotlinx.coroutines.selects.SelectInstance
```
Comment (freymend) [inline app/src/main/java/edu/uw/ischool/opensecrets/model/Entry.kt:3]:
```suggestion
```

Unused import

File: app/src/main/java/edu/uw/ischool/opensecrets/MainActivity.kt [hunk 11]
```diff
@@ -1,22 +1,51 @@
...
         } else {
 //            TODO: put home screen here
+
+            binding = ActivityMainBinding.inflate(layoutInflater)
+            setContentView(binding.root)
+            binding.searchButton.setOnClickListener {
+                startActivity(Intent(this, SearchActivity::class.java).addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP).addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION))
+            }
+            binding.addButton.setOnClickListener {
+                startActivity(Intent(this, EntryTextActivity::class.java).addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP).addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION))
```
Comment (freymend) [inline app/src/main/java/edu/uw/ischool/opensecrets/MainActivity.kt:44]: When I click on this button in the button nav, the app crashes

**Review by freymend — CHANGES_REQUESTED** [review]

File: app/src/main/java/edu/uw/ischool/opensecrets/MainActivity.kt (hunk 11 above)
Comment (duy007) [reply app/src/main/java/edu/uw/ischool/opensecrets/MainActivity.kt]: Can you send over the error message? I cannot seem to get this to trigger on my end.

Comment (duy007) [reply app/src/main/java/edu/uw/ischool/opensecrets/MainActivity.kt]: Removed color from XML that caused crashes.

File: app/build.gradle.kts
Comment (freymend) [inline app/build.gradle.kts:1]:
```diff
plugins {
    id("com.android.application")
    id("org.jetbrains.kotlin.android")
+ id("org.jetbrains.kotlin.kapt")
}
```
intellij complains if we don't use the kapt plugin with datBinding

**Review by freymend — APPROVED** [review]
